src/VisualStudio/Core/Def/StackTraceExplorer/StackFrameViewModel.cs

```diff
@@ -0,0 +1,310 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Documents;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.GoToDefinition;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.StackTraceExplorer;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.StackTraceExplorer
+{
+    internal class StackFrameViewModel : FrameViewModel
+    {
+        private readonly ParsedStackFrame _frame;
+        private readonly IThreadingContext _threadingContext;
+        private readonly Workspace _workspace;
+        private readonly IStreamingFindUsagesPresenter _streamingPresenter;
+
+        private ISymbol? _cachedSymbol;
+        private Document? _cachedDocument;
+        private int _cachedLineNumber;
+
+        public StackFrameViewModel(
+            ParsedStackFrame frame,
+            IThreadingContext threadingContext,
+            Workspace workspace,
+            IClassificationFormatMap formatMap,
+            ClassificationTypeMap typeMap,
+            IStreamingFindUsagesPresenter streamingPresenter)
+            : base(formatMap, typeMap)
+        {
+            _frame = frame;
+            _threadingContext = threadingContext;
+            _workspace = workspace;
+            _streamingPresenter = streamingPresenter;
+        }
+
+        public override bool ShowMouseOver => true;
+
+        public void NavigateToClass()
+        {
+            var cancellationToken = _threadingContext.DisposalToken;
+            Task.Run(() => NavigateToClassAsync(cancellationToken), cancellationToken);
+        }
+
+        public async Task NavigateToClassAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var symbol = await GetSymbolAsync(cancellationToken).ConfigureAwait(false);
+
+                if (symbol is not { ContainingSymbol: not null })
+                {
+                    // Show some dialog?
+                    return;
+                }
+
+                // Use the parent class instead of the method to navigate to
+                symbol = symbol.ContainingSymbol;
+
+                var sourceLocation = symbol.Locations.FirstOrDefault(l => l.IsInSource);
```

Comment:
Rather than getting the location directly this should use `ISymbolNavigationService`. That would make this work with Metadata as Source, and in future, SourceLink and embedded sources. Maybe thats a known follow up, i've forgotten :)


EDIT: Or `GoToDefinitionHelpers` like the below, I think does the right thing here.

Reply:
Fixed to have same navigation